### PR TITLE
[23] Upgrade test configuration to JavaSE-23

### PR DESCRIPTION
**Pull request by jarthana**
## What it does
Upgrades test configurations to JavaSE-23

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by jarthana** [comment]
OK, all tests are passing now, but hit the roadblock in https://github.com/eclipse-jgit/jgit/issues/36

Releasing this since the tests are green anyway.

**Comment by iloveeclipse** [comment]
@jarthana : the problem is
```
ERROR] Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:4.0.8:package-plugin (default-package-plugin) on project org.eclipse.jdt.debug.tests: /home/jenkins/agent/workspace/eclipse.jdt.debug-github_PR-482/org.eclipse.jdt.debug.tests/build.properties: bin.includes value(s) [java22_/] do not match any files. -> [Help 1]
```

File: org.eclipse.jdt.debug.tests/build.properties [hunk 1]
```diff
@@ -26,7 +26,8 @@ bin.includes = plugin.xml,\
                java8/,\
                java9/,\
                java16_/,\
-               java22/
+               java22_/,\
```
Comment (iloveeclipse) [inline org.eclipse.jdt.debug.tests/build.properties:29]: This line is the problem

Comment (jarthana) [reply org.eclipse.jdt.debug.tests/build.properties:29]: Thanks, which log did you find this in? Looks like the one I reported is a red herring then.

Comment (iloveeclipse) [reply org.eclipse.jdt.debug.tests/build.properties:29]: https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-482/16/console

**Comment by iloveeclipse** [comment]
There are two test failures:

[Test Result](https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-482/17/testReport/) (2 failures / +2)
[org.eclipse.jdt.debug.tests.launching.InstanceMainMethodsTests.testStaticMainWithoutArgs](https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-482/17/testReport/junit/org.eclipse.jdt.debug.tests.launching/InstanceMainMethodsTests/testStaticMainWithoutArgs/)
[org.eclipse.jdt.debug.tests.launching.InstanceMainMethodsTests.testDefaultMainWithoutArgs](https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-482/17/testReport/junit/org.eclipse.jdt.debug.tests.launching/InstanceMainMethodsTests/testDefaultMainWithoutArgs/)

**Comment by jarthana** [comment]
I am shooting in the dark here. Another factor is, we are running with a version of jdt.core (from master) with no support for Java 23 feature and that the execution doesn't produce expected result. The fact that we have added the compilation and source levels in master may be why there are no build path errors.

**Comment by jarthana** [comment]
As of now, looks like the maven central repo is inaccessible. Will wait for that to get resolved.

**Pull request merged**